core/server/master/src/test/java/alluxio/master/file/FileSystemMasterSyncMetadataMetricsTest.java

```diff
@@ -0,0 +1,705 @@
+import alluxio.util.io.PathUtils;
+
+import com.codahale.metrics.Counter;
+import com.google.common.collect.ImmutableList;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({UnderFileSystem.Factory.class})
+public class FileSystemMasterSyncMetadataMetricsTest {
+  private static final AlluxioURI ROOT = new AlluxioURI("/");
+  private static final String TEST_DIR_PREFIX = "/dir";
+  private static final String TEST_FILE_PREFIX = "/file";
+  @Rule
+  public TemporaryFolder mTempDir = new TemporaryFolder();
+  private String mUfsUri;
+  private FlakyLocalUnderFileSystem mUfs;
+  private ExecutorService mFileSystemExecutorService;
+  private ExecutorService mUfsStateCacheExecutorService;
+  private MasterRegistry mRegistry;
+  private DefaultFileSystemMaster mFileSystemMaster;
+
+  @Before
+  public void before() throws Exception {
+    UserState us = UserState.Factory.create(Configuration.global());
+    AuthenticatedClientUser.set(us.getUser().getName());
+
+    mTempDir.create();
+    mUfsUri = mTempDir.newFolder().getAbsolutePath();
+
+    mUfs = new FlakyLocalUnderFileSystem(new AlluxioURI(mUfsUri),
+        UnderFileSystemConfiguration.defaults(Configuration.global()));
+    PowerMockito.mockStatic(UnderFileSystem.Factory.class);
+    Mockito.when(UnderFileSystem.Factory.create(anyString(), any())).thenReturn(mUfs);
+
+    Configuration.set(PropertyKey.MASTER_JOURNAL_TYPE, JournalType.UFS);
+    Configuration.set(PropertyKey.MASTER_MOUNT_TABLE_ROOT_UFS, mUfsUri);
+    String journalFolderUri = mTempDir.newFolder().getAbsolutePath();
+
+    mFileSystemExecutorService = Executors
+        .newFixedThreadPool(4, ThreadFactoryUtils
+            .build("FileSystemMaster-%d", true));
+    mUfsStateCacheExecutorService = Executors
+        .newFixedThreadPool(4, ThreadFactoryUtils
+            .build("UfsStateCache-%d", true));
+    mRegistry = new MasterRegistry();
+    JournalSystem journalSystem =
+        JournalTestUtils.createJournalSystem(journalFolderUri);
+    CoreMasterContext context = MasterTestUtils.testMasterContext(journalSystem);
+    new MetricsMasterFactory().create(mRegistry, context);
+    BlockMaster blockMaster = new BlockMasterFactory().create(mRegistry, context);
+    mFileSystemMaster = new DefaultFileSystemMaster(blockMaster, context,
+        ExecutorServiceFactories.constantExecutorServiceFactory(mFileSystemExecutorService));
```

Comment:
double check this thread pool is closed on `after()`

Reply:
sure
https://github.com/Alluxio/alluxio/blob/c5e27f9c2dd5056e384c44df4730d60acb6dc04b/core/server/master/src/test/java/alluxio/master/file/FileSystemMasterSyncMetadataMetricsTest.java#L129-L133